controller/QtiCreator.php

```diff
@@ -348,4 +360,152 @@ private function validateXmlInput(string $xml)
         }
         \tao_helpers_Xml::getSimpleXml($xml);
     }
+
+    /**
+     * Download sample template for tabular import
+     *
+     * @param string 
+     * @throws common_exception_Error
+     */
+    public function downloadCsv()
```

Comment:
Do you think all this logic of building the template, collection and parsing the metadata, populating the lines, etc should be inside the controller?

Reply:
Done